devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
         return value
+
+    if isinstance(value, basestring):
+        value = value.strip()
+        if value.endswith('%'):
+            try:
+                return float(value.rstrip('%')) / 100
```

Comment:
Yeah that's true ...  What I was thinking is that having `numeric()` working on list might be ok (if it's documented like that at least). Anyway the above code does not suffice at all for that, and yeah, the code simplification I was aiming at with `ast.eval_literal()` is actually ruined by all the additional checks we need to add to make it work properly. So forget about it, I'm OK with your implementation. Sorry for bothering you :smile: